toast-tk-web-agent/src/main/java/io/toast/tk/agent/ui/panels/AbstractPanel.java

```diff
@@ -0,0 +1,177 @@
+		Dimension prefDim = new Dimension(textField.getPreferredSize().width, 30);
+		Dimension moyDim = new Dimension(textField.getMaximumSize().width, 40);
+		textField.setPreferredSize(prefDim);
+        textField.setMaximumSize(moyDim);
+        textField.setFont(PanelHelper.FONT_TEXT);
+		textField.setLayout(new BorderLayout());
+		textField.setColumns(30);
+		textField.setAlignmentX(Component.LEFT_ALIGNMENT);	
+		textField.setAlignmentY(Component.CENTER_ALIGNMENT);
+		textField.addKeyListener(new KeyListener() {
+			@Override
+			public void keyPressed(KeyEvent a) {
+				if (a.getKeyCode() == KeyEvent.VK_ENTER) {
+					try {
+						testIconValid(false);
+					} catch (IOException e) {
+						LOG.error(e.getMessage(), e);
+					}
+				}
+			}
+
+			@Override
+			public void keyReleased(KeyEvent e) {
+			}
+
+			@Override
+			public void keyTyped(KeyEvent e) {
+			}
+		});
+		
+		return textField;
+	}
+
+	public abstract void testIconValid(boolean retry) throws IOException;
+
+	public JLabel createIconValid() {
+		JLabel iconValid = new JLabel(new ImageIcon(this.validImage));
+		return iconValid;
+	}
+	
+	public JLabel createIconNotValid() {
+		JLabel iconNotValid = new JLabel(new ImageIcon(this.notvalidImage));
+		return iconNotValid;
+	}
+
+	protected abstract JComponent buildPanel() throws IOException;
+
+	private void buildContentPanel() throws IOException{	
+		this.add(Box.createHorizontalGlue());
+		this.setLayout(new BoxLayout(this, BoxLayout.PAGE_AXIS));
+		this.setAlignmentX(Component.CENTER_ALIGNMENT);
+		this.setAlignmentY(Component.CENTER_ALIGNMENT);
+		this.setBackground(Color.white);
+		
+		JPanel textButtonPanel = PanelHelper.createBasicPanel(BoxLayout.LINE_AXIS);
+
+		textField = createBasicTextPanel();
+		iconPanel = PanelHelper.createBasicPanel();
+		iconValid = createIconValid();
+		iconNotValid = createIconNotValid();
+
+		JComponent component = buildPanel();
+		if(component != null){
+			textButtonPanel.add(component);
+		}
+
+		textButtonPanel.add(textField);
+		textButtonPanel.add(iconPanel);
+
+		textButtonPanel.setSize(new Dimension(textButtonPanel.getSize().width, 50));
+		textButtonPanel.setMaximumSize(new Dimension(textButtonPanel.getMaximumSize().width, 50));
+		
+		this.add(textButtonPanel);
+		this.add(errorLabel);
+	}
+	
+	protected JLabel buildErrorLabel(JPanel iconPanel, EnumError testValue) throws IOException{
+		JLabel iconValid = new JLabel(new ImageIcon(this.validImage));
+		JLabel iconNotValid = new JLabel(new ImageIcon(this.notvalidImage));
+		iconValid.setBackground(Color.white);
+		iconNotValid.setBackground(Color.white);
+		String errorMessage = " ";
+		if(testValue == EnumError.FILE || testValue == EnumError.DIRECTORY ||testValue == EnumError.URL) {
+			iconPanel.add(iconValid);		
+		} else if(testValue == EnumError.APIKEY) { // apiKey does not have verification
+			errorMessage = errorMessageApiKey;
+			JLabel toastLogo = new JLabel(new ImageIcon(this.toastLogo));
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Rename "toastLogo" which hides the field declared at line 37. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AHiddenFieldCheck)